class.jetpack-gutenberg.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+/**
+ * General Gutenberg editor specific functionality
+ */
+class Jetpack_Gutenberg {
+	/**
+	 * Check if Gutenberg editor is available
+	 *
+	 * @since 6.7.0
+	 *
+	 * @return bool
+	 */
+	public static function is_gutenberg_available() {
+		return function_exists( 'register_block_type' );
+	}
+
+	/**
+	 * Load Gutenberg assets
+	 *
+	 * @since 6.7.0
+	 *
+	 * @return void
+	 */
+	public static function enqueue_block_assets() {
+		if ( ! self::should_load_blocks() ) {
+			return;
+		}
+
+		$rtl = is_rtl() ? '.rtl' : '';
+
+		/**
+		 * Filter to enable serving blocks via CDN
+		 *
+		 * CDN cache is busted once a day or when Jetpack version changes. To customize it:
+		 *   add_filter( 'jetpack_gutenberg_cdn_cache_buster', function( $version ) { return time(); }, 10, 1 );
+		 *
+		 * @since 6.5.0
+		 *
+		 * @param bool false Whether to load Gutenberg blocks from CDN
+		 */
+		if ( apply_filters( 'jetpack_gutenberg_cdn', false ) ) {
+			$cdn_base    = 'https://s0.wp.com/wp-content/mu-plugins/jetpack/_inc/blocks';
+			$view_script = "$cdn_base/view.js";
+			$view_style  = "$cdn_base/view$rtl.css";
+
+			/**
+			 * Filter to modify cache busting for Gutenberg block assets loaded from CDN
+			 *
+			 * @since 6.5.0
+			 *
+			 * @param string
+			 */
+			$version = apply_filters( 'jetpack_gutenberg_cdn_cache_buster', sprintf( '%s-%s', gmdate( 'd-m-Y' ), JETPACK__VERSION ) );
+		} else {
+			$view_script = plugins_url( '_inc/blocks/view.js', JETPACK__PLUGIN_FILE );
+			$view_style  = plugins_url( "_inc/blocks/view$rtl.css", JETPACK__PLUGIN_FILE );
+			$version     = Jetpack::is_development_version() && file_exists( JETPACK__PLUGIN_DIR . '_inc/blocks/view.js' )
+				? filemtime( JETPACK__PLUGIN_DIR . '_inc/blocks/view.js' )
+				: JETPACK__VERSION;
+		}
+
+		wp_enqueue_script( 'jetpack-blocks-view', $view_script, array(), $version );
+		wp_enqueue_style( 'jetpack-blocks-view', $view_style, array(), $version );
+	}
+
+	/**
+	 * Load Gutenberg editor assets
+	 *
+	 * @since 6.7.0
+	 *
+	 * @return void
+	 */
+	public static function enqueue_block_editor_assets() {
+		if ( ! self::should_load_blocks() ) {
+			return;
+		}
+
+		$rtl = is_rtl() ? '.rtl' : '';
+
+		/**
```

Comment:
I would recommend only adding the docblock once. It makes it easier to see that this is not the only occurrence of the filter in the codebase, and it makes our doc parser happy. :)

You can replace that second occurrence by a quick comment like `/** This filter is already documented above */`

Reply:
👍 5d106ecdd5d5d2d2549391083fde01b46a805a1b